vscode/src/commands.ts

```diff
@@ -11,36 +14,79 @@ function getFullScreenTab() {
 }
 
-const commandsMap: (
-  extensionContext: vscode.ExtensionContext,
-  state: ExtensionState,
-) => {
+async function analyzeFileContent(contentString: string, state: ExtensionState) {
+  const { extensionContext } = state;
+
+  const incidentDataString = extensionContext.workspaceState.get<string>("incidentData", "[]");
+  const incidentData = JSON.parse(incidentDataString) as Incident[];
+
+  const diagnosticCollection = vscode.languages.createDiagnosticCollection("konveyor");
+  const diagnosticsMap: Map<string, vscode.Diagnostic[]> = new Map();
+
+  incidentData.forEach((incident) => {
+    const fileUri = vscode.Uri.parse(incident.file);
+
+    const lineNumber = incident.line ? incident.line - 1 : 0;
+    const severity =
+      incident.severity === "High"
+        ? vscode.DiagnosticSeverity.Error
+        : incident.severity === "Medium"
+          ? vscode.DiagnosticSeverity.Warning
+          : vscode.DiagnosticSeverity.Information;
+
+    const diagnostic = new vscode.Diagnostic(
+      new vscode.Range(
+        new vscode.Position(lineNumber, 0),
+        new vscode.Position(lineNumber, Number.MAX_VALUE),
+      ),
+      incident.message,
+      severity,
+    );
+
+    const diagnostics = diagnosticsMap.get(fileUri.toString()) || [];
+    diagnostics.push(diagnostic);
+    diagnosticsMap.set(fileUri.toString(), diagnostics);
+  });
+
+  diagnosticsMap.forEach((diagnostics, fileUri) => {
+    diagnosticCollection.set(vscode.Uri.parse(fileUri), diagnostics);
+  });
+
+  const sidebarProvider = state.sidebarProvider;
+  if (sidebarProvider && sidebarProvider.webview) {
+    sidebarProvider.webview.postMessage({
+      command: "incidentData",
+      data: incidentData,
+    });
+  }
+
+  // Optionally, show an information message upon completion
+  vscode.window.showInformationMessage("Diagnostics created based on mock analysis.");
+}
+
+const commandsMap: (state: ExtensionState) => {
   [command: string]: (...args: any) => any;
-} = (extensionContext, state) => {
-  const { sidebarProvider } = state;
+} = (state) => {
+  const { sidebarProvider, extensionContext } = state;
   return {
     "konveyor.startAnalysis": async (resource: vscode.Uri) => {
       if (!resource) {
         vscode.window.showErrorMessage("No file selected for analysis.");
         return;
       }
 
-      // Get the file path
       const filePath = resource.fsPath;
 
-      // Perform your analysis logic here
       try {
-        // For example, read the file content
         const fileContent = await vscode.workspace.fs.readFile(resource);
         const contentString = Buffer.from(fileContent).toString("utf8");
 
-        console.log(contentString, fileContent);
-
-        // TODO: Analyze the file content
-        vscode.window.showInformationMessage(`Analyzing file: ${filePath}`);
+        await extensionContext.workspaceState.update(
+          "incidentData",
+          JSON.stringify(generateMockIncidentData()),
+        );
```

Comment:
My original thought was that whatever process performs analysis would pass `[]Ruleset` ([found here](https://github.com/konveyor/analyzer-lsp/blob/8f5ccf72040d7d89765aa3b8fc86e8d7cb4be684/output/v1/konveyor/violations.go#L16))as a parameter, and do the parsing there.

I'm wondering if you are thinking it would work differently...maybe the analysis process would pass the `[]RuleSet` to a "process function" that put it in the workspace state where it got added to diagnostics (etc. etc.).

Reply:
This definitely help me understand what we are looking for. Need to think on it a bit more but I can make the change and have the ruleset be passed in.